internal/controllers/core/cluster/cache_fake.go

```diff
@@ -0,0 +1,70 @@
+package cluster
+
+import (
+	"sync"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/tilt-dev/tilt/internal/k8s"
+	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
+)
+
+type FakeClientCache struct {
```

Comment:
this might be a dumb question, but why not use the real cache? i feel like it exposes the same interface

Reply:
The cluster package's implementation of `ClientCache` is `ConnectionManager`, which has the full `connection` object/state to avoid having to do book-keeping 2x (state + clients), so it's not usable as a generic implementation. That's definitely refactorable though!

Maybe this should just be called `ClientMap` or something, which is a generic/naive impl used for tests and then the cluster reconciler can still `wire.Bind` its specialized version for prod 🤔

Reply:
re: "book-keeping 2x (state + clients)", what bookkeeping are you talking about exactly?

Reply:
The client-or-error vs everything else in `connection`:
```go
type connection struct {
	connType     connectionType
	spec         v1alpha1.ClusterSpec
	dockerClient docker.Client
	k8sClient    k8s.Client
	error        string
	createdAt    time.Time
	arch         string
}
```

Currently, the `Cluster` reconciler uses the `ConnectionManager` type to maintain its internal state (latest processed spec, created at, arch) + the client, and since that implements/is wire-bound to `ClientCache` it lets consumers get the client out without leaking other the other details, which should be read via API.

The "fake" only stores the client-or-error since it's not tied to the reconciler. Again, name is a sliiiight misnomer since it doesn't really have a lot of specialized test code in it (besides the ability to pre-seed a default client)

I can definitely split it though and have something more like:
```
type Reconciler struct {
   ...
   clients ClientCache
   state    map[types.NamespacedName]connection
}
```

Reply:
i think we're talking past each other. here's a commit that demonstrates what i'm suggesting:
https://github.com/tilt-dev/tilt/commit/c05bac10c9f09fc32f8566d7ef617079c24d7b64

Reply:
Ahhhh ok yes now I understand what you meant!

I'll pull that over 👍